tests/test_visitors/test_ast/test_subscripts/test_slice_assignment.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from wemake_python_styleguide.violations.consistency import (
+    AssignToSliceViolation,
+)
+from wemake_python_styleguide.visitors.ast.subscripts import SubscriptVisitor
+
+usage_template = 'constant[{0}]= [1,2,3]'
+
+
+@pytest.mark.parametrize('expression', [
+    ':7',
+    '1:7:2',
+    '3:',
+    '3::',
+    '::2',
+    ':2:',
```

Comment:
Let's also test this case:

```python
>>> l = [1, 2, 3]
>>> l[slice(1)]
[1]
```

Reply:
Thanks will take care of that. 
@sobolevn  Also wanted to check do we allow something like this? replacing the full list or forbid that as well?
```
>>> l = [1, 2, 3]
>>> l[:] = [4, 5]
```

Reply:
Should be banned as well!